File: src/SIR.Simulation/Substrate.fs
namespace SIR.Simulation

open FS.GG.Game.Core
open SIR.Domain

/// The minimal published-package seam used before the first simulation slice.
[<RequireQualifiedAccess>]
module Substrate =
    let cellBytes (cell: Cell) =
        CanonicalEncoding.concatenate
            [ CanonicalEncoding.int32LittleEndian cell.Col
              CanonicalEncoding.int32LittleEndian cell.Row ]

    let origin = { Col = 0; Row = 0 }

File: src/SIR.Client.Web/.fable-rules/SIR.Simulation/Substrate.js

import { int32LittleEndian, concatenate } from "../SIR.Domain/CanonicalEncoding.js";
import { Cell } from "../fable_modules/FS.GG.Game.Core.0.13.0/Primitives.fs.js";

export function cellBytes(cell) {
    return concatenate([int32LittleEndian(cell.Col), int32LittleEndian(cell.Row)]);
}

export const origin = new Cell(0, 0);

